src/isoStringScalar/isoStringScalar.ts

```diff
@@ -0,0 +1,93 @@
+import { GraphQLScalarType, Kind } from 'graphql';
+import { DateTime } from 'luxon';
+import {
+  InternalServerError,
+  UserInputError,
+} from '../errorHandler/errorHandler';
+
+export const isoStringScalar = new GraphQLScalarType({
+  name: 'ISOString',
+  description: 'ISOString custom scalar type',
+
+  /**
+   * Converts server-side TS Date representation to a JSON-compatible, ISO-8601-compliant
+   * UTC Datetime String for Apollo Server to include in an operation response.
+   *
+   * @param value - TS Date Object as generated by Data Store Client & validated by mysqlDateConvert.
+   * @returns ISO-8601-compliant, UTC-based Datetime String.
+   */
+  serialize(value: Date | null): string | null {
+    if (value === null) {
+      return null;
+    }
+
+    if (!(value instanceof Date)) {
+      throw new InternalServerError(
+        'GraphQL ISOString Scalar serializer expected a `Date` object or null'
+      );
+    }
+
+    // isNaN here checks for 0000-00-00 style dates, which are invalid
+    if (isNaN(value.valueOf())) {
+      throw new InternalServerError(
```

Comment:
I think this is where we want to return `null` based on the other conversation.

Reply:
(this then also covers our butts if the data layer ends up passing other invalid values beyond 0000-00-00)

Reply:
I'm in a mixed mind.

I agree that would be more ideal. If we were not coping with prior precedent, I would also try to push the hard line to avoid eroding confidence in the first place.

It definitely isn't ideal to arrive here with a NaN Date, and I don't think we want that to be our default pattern in the servers. I think ideally we would just store null in the database and side-step this entire need for learned convention and introduction of ambiguity. We do need to cope with that convention though. If the schema allows the date to be nulled, it is probably a better client experience to return null and allow the rest of the data to make it through. If the date is not nullable, that will result in a schema level error.

How about returning null and triggering a manual sentry error if it is available in the running service?

Reply:
@hyperparabolic specifically line 20-21 https://github.com/Pocket/apollo-utils/pull/220/files#diff-f1984233552213cafd62444ab2590ed293502d4213be0c3ec2f5a7586d45f289R20

@cmharlow maybe it would make the control flow more obvious to use a `switch/case` vs. `ifs`

Reply:
guys, if you're coing to change the scope here beneath me, lets at least zoom about it. What you're proposing here **is not** what was laid out or has been laid out.

Reply:
@kschelonka are you saying I should use a switch instead of ifs here? or you just prefer it?

Reply:
discussed in slack

Reply:
Sorry for injecting some uncertainty here. I wasn't sure if we needed to guard for that or not. I think this is good.